docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,912 @@
+performance and gather feedback.
+
+**Resources & Prerequisites**:
+
+- Completed and tested configuration from [Step 3](#step-3---iterative-implementation--testing).
+- Access to Iroha 2 codebase and integration tools.
+- A pool of testers or users for field-testing.
+
+**Deliverables**:
+
+- Configuration successfully integrated into Iroha 2.
+- Feedback report from field tests detailing user experience, potential issues, and improvements.
+
+**Potential Risks**:
+
+- Integration challenges due to unforeseen incompatibilities.
+- Negative user feedback or usability challenges.
+
+**Feedback Loop**:
+
+- Regular integration testing and validation.
+- Direct communication channels with field testers to gather timely feedback.
+
+### Step 5 - Iterative Refinement & Optimization
+
+**Description**: After integrating the new configuration into Iroha 2, there might be some elements that need refining
+based on user feedback and field-testing.
+
+**Resources & Prerequisites**:
+
+- Feedback from the [integration phase](#step-4---integration-field-testing--feedback-collection).
+- Development and testing environments.
+
+**Deliverables**:
+
+- A refined configuration addressing user feedback.
+- Documentation updates reflecting changes made.
+
+**Potential Risks**:
+
+- Over-optimization leading to reduced clarity.
+- Potential rework if feedback indicates major design flaws.
+
+**Feedback Loop**:
+
+- Continued consultation with users.
+- Performance isn't a primary concern, but the configuration should be robust and user-friendly.
+
+## Impact Assessment
+
+The proposed overhaul of Iroha 2's configuration system stands to significantly impact various facets of the project,
+both technical and user-facing. These implications touch on the project's future scalability, user experience, and
+maintainability.
+
+### Technical Impact
+
+- **Code Complexity**: The pursuit of an enhanced user experience may lead to a more intricate configuration codebase.
+  This increased complexity might make the system harder to maintain and debug.
+- **Performance Impact**: While the configuration happens at startup and isn't a recurring process, there might be a
+  slight performance dip due to the enhancements. This is considered acceptable given the trade-off for improved user
+  experience.
+
+### Operational Impact
+
+- **User Transition**: Given the main goal of this RFC — simplifying the configuration for end users — we anticipate a
+  minimal migration pain for current users. The proposed changes, informed by earlier sections of this RFC, are intended
+  to streamline and simplify the process for them.
+- **Deployment & Migration**: The revamped configuration could bring new deployment methods or procedures. Clear
+  documentation will be essential to guide system administrators and users through these changes.
+
+### User Impact
+
+- **Enhanced User Experience**: The main drive behind these changes is to offer users a more intuitive configuration
```

Comment:
Do we need some sort of tool for easy converting from old config to the new one?

Reply:
Probably. Either a tool or a migration guide. I believe it is not a major point, since we don't have many users as of now, but it is worth it note this in the RFC.

Reply:
Addressed: https://github.com/hyperledger/iroha-rfcs/pull/3/commits/f89aff51901362918e395c081d3c88132bc3a39d